### PR TITLE
Fixes missing association admin class in datagrid filters.

### PR DESCRIPTION
**Pull request by jarkt**
Related to this problem: https://stackoverflow.com/questions/38440152/autocomplete-filter-in-sonata-admin-mongodb

I am targeting this branch, because the change is backward compatible.

## Changelog

```markdown
### Fixed
Fixes missing association admin class in datagrid filters.
```

### CODE REVIEW

File: src/Builder/DatagridBuilder.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 namespace Sonata\DoctrineMongoDBAdminBundle\Builder;
 
+use Doctrine\ODM\MongoDB\Mapping\ClassMetadataInfo;
```
Comment (kunicmarko20) [inline src/Builder/DatagridBuilder.php]: cc @greg0ire wasn't `ClassMetadataInfo` deprecated?

Comment (greg0ire) [reply src/Builder/DatagridBuilder.php]: The ORM one is.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased]
+### Fixed
+- Fixes missing association admin class in datagrid filters.
+
```
Comment (greg0ire) [inline CHANGELOG.md]: Don't touch this file directly please, we do it on release because this is a recipe for conflicts

Comment (jarkt) [reply CHANGELOG.md]: Updated.

File: src/Builder/DatagridBuilder.php (hunk 1 above)
Comment (greg0ire) [reply src/Builder/DatagridBuilder.php]: But maybe the one in ODM is too, there is https://github.com/doctrine/mongodb-odm/blob/1.3.x/lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php#L40

Comment (greg0ire) [reply src/Builder/DatagridBuilder.php]: @alcaeus can you please advise?

Comment (alcaeus) [reply src/Builder/DatagridBuilder.php]: Correct - this will also be dropped in ODM 2.0.

Comment (jarkt) [reply src/Builder/DatagridBuilder.php]: But it's only deprecated from release 1.3 and the current one is 1.2.5. There is no ClassMetadata in current release.

Comment (greg0ire) [reply src/Builder/DatagridBuilder.php]: Ah right. Let's keep it that way then.

**Comment by greg0ire** [comment]
Please run `make cs-fix`

File: src/Builder/DatagridBuilder.php (hunk 1 above)
Comment (alcaeus) [reply src/Builder/DatagridBuilder.php]: > But it's only deprecated from release 1.3 and the current one is 1.2.5.

Technically correct.

> There is no ClassMetadata in current release.

Yes there is: https://github.com/doctrine/mongodb-odm/blob/b47979e8af5830f15790ea731441e4bdf6876d80/lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php#L40

Comment (jarkt) [reply src/Builder/DatagridBuilder.php]: Crazy, don't know why I've not seen it. I think I switched my branch faster than my ide has updated. ;)

**Comment by jarkt** [comment]
I've "make cs-fix" let run. I don't know why it has not already run before. So there are changes which aren't mine. But ok, there is a new commit now.

**Comment by greg0ire** [comment]
> I don't know why it has not already run before.

You're the first to contribute since this configuration is enabled

**Comment by greg0ire** [comment]
Thanks @jarkt !

**Pull request merged**